Date: Aug 31, 2016 12:46 PM

From: e4471ee03f0ed43bb17107373556e45a2cb0d129 Sep 15, 2016 1:52 AM

Subject: [PATCH] RDKB-6774,RDKB-6775,RDKB-7628: lighttpd to 1.4.39

	RDKB-6774 upgrading lighttpd to 1.4.39, RDKB-6775 mod_csrf changes for lighttpd_1.4.39.

Source: COMCAST

Upstream-Status: Pending

Signed-off-by: rthangamani <ramakrishnan.thangamani@wipro.com>
---

Index: lighttpd-1.4.39/src/network.c
===================================================================
--- lighttpd-1.4.39.orig/src/network.c
+++ lighttpd-1.4.39/src/network.c
@@ -273,6 +273,20 @@
 		goto error_free_socket;
 	}
 
+#ifdef SO_BINDTODEVICE
+        /* host might be an interface name, so try to bind to it */
+        if (host && (srv_socket->addr.plain.sa_family == AF_INET
+#ifdef HAVE_IPV6
+                     || srv_socket->addr.plain.sa_family == AF_INET6)) {
+#else
+                    )) {
+#endif
+                if (!setsockopt(srv_socket->fd, SOL_SOCKET, SO_BINDTODEVICE,
+                                host, strlen(host)))
+                        host = NULL;
+        }
+#endif
+
 	switch(srv_socket->addr.plain.sa_family) {
 #ifdef HAVE_IPV6
 	case AF_INET6:
